kdf.go

```diff
@@ -27,20 +29,30 @@ in RFC 5054 §2.6, which says
     x = SHA1(s | SHA1(I | ":" | P))
 */
 func KDFRFC5054(salt []byte, username string, password string) (x *big.Int) {
-
 	p := []byte(PreparePassword(password))
 
 	u := []byte(PreparePassword(username))
 
 	innerHasher := sha1.New() // #nosec
-	innerHasher.Write(u)
-	innerHasher.Write([]byte(":"))
-	innerHasher.Write(p)
+	if _, err := innerHasher.Write(u); err != nil {
+		panic(err)
+	}
+	if _, err := innerHasher.Write([]byte(":")); err != nil {
+		panic(err)
+	}
+	if _, err := innerHasher.Write(p); err != nil {
+		panic(err)
+	}
```

Comment:
I choose to panic, rather than to error because I want to keep this PRs scope to the minumin (just making the linters pass). I'll propose in a separate PR changes to some of the exported funcs, as I believe they should be returning errors